redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewBox.kt

```diff
@@ -38,83 +40,88 @@ import app.cash.redwood.widget.ViewGroupChildren
 internal class ViewBox(
   context: Context,
 ) : FrameLayout(context), Box<View> {
+  private val density = Density(context.resources)
+  private var horizontalAlignment = CrossAxisAlignment.Start
+  private var verticalAlignment = CrossAxisAlignment.Start
+  private var width = Constraint.Wrap
+  private var height = Constraint.Wrap
+
   override var modifier: Modifier = Modifier
 
   override val value get() = this
 
   override val children = ViewGroupChildren(this)
 
-  private val density = Density(context.resources)
-
-  private var defaultHorizontalAlignment = CrossAxisAlignment.Start
-  private var defaultVerticalAlignment = CrossAxisAlignment.Start
-  private var width = Constraint.Fill
-  private var height = Constraint.Fill
-
-  init {
-    layoutParams = LayoutParams(MATCH_PARENT, MATCH_PARENT)
+  override fun generateDefaultLayoutParams(): LayoutParams {
+    return LayoutParams(WRAP_CONTENT, WRAP_CONTENT)
   }
 
   override fun width(width: Constraint) {
     this.width = width
+    invalidate()
   }
 
   override fun height(height: Constraint) {
     this.height = height
+    invalidate()
   }
 
   override fun margin(margin: Margin) {
-    updateLayoutParams {
-      val layoutParams = this as MarginLayoutParams
+    updateLayoutParams<MarginLayoutParams> {
       with(density) {
-        layoutParams.setMargins(
-          margin.start.toPxInt(),
-          margin.top.toPxInt(),
-          margin.end.toPxInt(),
-          margin.bottom.toPxInt(),
+        updateMargins(
```

Comment:
Should be `updateMarginsRelative` with `start` and `end`, I think.

Going to file an issue to add RTL layout to our tests (not that we do a good job of even exercising margins in tests today).

Reply:
Looks like we have them for flex containers already. So I'll just note it on the Box issue.